test/functional/job/job_spec.lua

```diff
@@ -188,6 +189,41 @@ describe('jobs', function()
     eq({'notification', 'exit', {45, 10}}, next_msg())
   end)
 
+  it('cant redefine callbacks being used by a job', function()
+    local screen = Screen.new()
+    screen:attach()
+    local script = [[
+      function! g:JobHandler(job_id, data, event)
+      endfunction
+
+      let g:callbacks = {
+      \ 'on_stdout': function('g:JobHandler'),
+      \ 'on_stderr': function('g:JobHandler'),
+      \ 'on_exit': function('g:JobHandler')
+      \ }
+      let job = jobstart('cat -', g:callbacks)
```

Comment:
When we run tests on windows, it might be a good a idea to use msys to provide a minimal set of unix command line tools